### PR TITLE
Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by kwambiee**
1. Created classes and objects with ruby
2. Implemented inheritance and encapsulation with ruby

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Student < Person
+  def initialize(classroom)
+    super('Joy', 24, false)
```
Comment (zhadier) [inline student.rb:3]: - [x] Great work with all of the logic. But one thing that I kindly require you to do is to add the `age` `permission` and `name` as parameters to Student aswell. That way when you create a Student, you'll be able to define all these attributes for the super `person` aswell. The code will be something like
``` ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (zhadier) [inline teacher.rb:5]: - [ ] Kindly do the same as I requested in the Student initializer

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @kwambiee  👋 

Good job so far! 🌤️ 
There are some issues that you still need to work on to go to the next project but you are almost there! 🍓 



## Highlights
✔️ No linter errors
✔️ GitFlow was set up correctly
✔️ Professional pull request title and description

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 

### **Optional**
- [ ] Hello I don't see any changes, and I see the app is returning the correct values for each file but as the previous CR said you should replace the values that you have added in the super and replace them with the argument that he left you as hints in that way the app in the next milestone instead of putting the values and printing them we will be able to create them from the terminal.

- [ ] Please remove lines `P1 and S1 and T1 from the files since they are not necessary.`
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**